### PR TITLE
TST: Update of_site doctest result

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to update `of_site` doctest result due to new merges into astropy-data.

Without this patch, you will see doctest error like this:

```
_____________________________ [doctest] index.rst ______________________________
331 
332     >>> from astropy.coordinates import EarthLocation
333     >>> apo = EarthLocation.of_site('Apache Point Observatory')
334     >>> apo  # doctest: +FLOAT_CMP
335     <EarthLocation (-1463969.30185172, -5166673.34223433, 3434985.71204565) m>
336 
337 To see the list of site names available, use
338 :func:`~astropy.coordinates.EarthLocation.get_site_names`::
339 
340     >>> EarthLocation.get_site_names()  # doctest: +REMOTE_DATA
Expected:
    ['ALMA', 'AO', 'ARCA', 'ASKAP', 'ATST', 'Anderson Mesa', ...]
Got:
    ['ALMA', 'AO', 'ARCA', 'ASKAP', 'ATA', 'ATST', 'Allen Telescope Array', 'Anderson Mesa', (and so on)
```

Example log: https://github.com/astropy/astropy/actions/runs/3511169622/jobs/5881686779

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by hamogu** [comment]
@dlakaplan  This is related. It's rare that updates to sites.json break astropy, but I guess it can happen if the particular example added gives and acronym very early in the alphabet...

**Comment by pllim** [comment]
The convolution failures are not related (yay Monday).

**Review by hamogu — COMMENTED** [review]
This is from https://github.com/astropy/astropy-data/pull/92 which added and observatory that just happens to appear early on an alphabetical listing.

This is good to go - I'm just been told off for approving PRs before CI passes, so I'm only doing "comment" right now, until all steps have at least run (even if some have unrealted failures).

**Comment by pllim** [comment]
I was working off an outdated failure log, so I re-pushed the commit with hopefully the correct fix.

File: docs/coordinates/index.rst [hunk 1]
```diff
@@ -338,7 +338,7 @@ To see the list of site names available, use
 :func:`~astropy.coordinates.EarthLocation.get_site_names`::
 
     >>> EarthLocation.get_site_names()  # doctest: +REMOTE_DATA
-    ['ALMA', 'AO', 'ARCA', 'ASKAP', 'ATST', 'Anderson Mesa', ...]
+    ['ALMA', 'AO', 'ARCA', 'ASKAP', 'ATA', ATST', ...]
```
Comment (eerovaher) [inline docs/coordinates/index.rst:341]: We can reduce the chances of the expected output changing if we write out less names:
```suggestion
    ['ALMA', 'AO', 'ARCA', ...]
```

**Comment by pllim** [comment]
I had a typo anyway, so I accepted @eerovaher 's suggestion. I think even with 3, the doc gets the point across. Should I squash?

**Comment by pllim** [comment]
Thanks for the reviews!

**Comment by pllim** [comment]
@meeseeksdev backport to v5.2.x

**Comment by pllim** [comment]
This line does not exist in v5.0.x, so I updated the milestone. FYI.

**Pull request merged**